cmd/zpool/zpool_main.c

```diff
@@ -2357,8 +2358,32 @@ health_str_to_color(const char *health)
 	return (NULL);
 }
 
+/*
+ * If printing out only unhealthy vdevs, then loop through children so that
+ * if any of them are unhealthy, we print our own line as a parent.
+ */
+static boolean_t
+check_child_health(nvlist_t *nv, uint_t children)
+{
+	/* Loop through children and return true on first error found */
+	uint_t c, vsc;
+	vdev_stat_t *vs;
+
+	for (c = 0; c < children; c++) {
```

Comment:
If I'm not missing something, as `c` isn't being used in this loop, so why not use a `do..while` loop instead?
It can possibly save a few instructions.

Reply:
Yea, it looks like the intention here is to iterate over all the child vdevs:
```c
        (void) nvlist_lookup_nvlist_array(nvroot, ZPOOL_CONFIG_CHILDREN,
            &children_nv_array, &num_children);
         ...
         for (c = 0; c < num_children; c++) {
             nvlist_t *nv = children_nv_array[c];
         ...
 ```

Reply:
Thank you @rilysh. You and @tonyhutter helped reveal a logic flaw. New code does retain (and use) the 'c' counter.